secboot/keys/plainkey_test.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nosecboot
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package keys_test
+
+import (
+	"bytes"
+	"crypto"
+	"io"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	sb "github.com/snapcore/secboot"
+
+	"github.com/snapcore/snapd/secboot/keys"
+)
+
+type plainkeySuite struct {
+}
+
+var _ = Suite(&plainkeySuite{})
+
+func (s *plainkeySuite) SetUpTest(c *C) {
+}
+
+type MyKeyDataWriter struct {
+	*bytes.Buffer
+}
+
+func NewMyKeyDataWriter() *MyKeyDataWriter {
+	return &MyKeyDataWriter{
+		Buffer: bytes.NewBuffer([]byte{}),
+	}
+}
+
+func (kdw *MyKeyDataWriter) Commit() error {
+	return nil
+}
+
+func (s *plainkeySuite) TestPlainKey(c *C) {
+	restore := keys.MockSbNewProtectedKey(func(rand io.Reader, protectorKey []byte, primaryKey sb.PrimaryKey) (protectedKey *sb.KeyData, primaryKeyOut sb.PrimaryKey, unlockKey sb.DiskUnlockKey, err error) {
+		kd, err := sb.NewKeyData(&sb.KeyParams{
+			Handle:       nil,
+			Role:         "run",
+			PlatformName: "fakePlatform",
+			KDFAlg:       crypto.SHA256,
+		})
+		c.Assert(err, IsNil)
+		return kd, primaryKey, []byte("unlock-key"), nil
+	})
+	defer restore()
+
+	protectorKey, err := keys.NewProtectorKey()
+	c.Assert(err, IsNil)
+	protectedKey, primaryKeyOut, unlockKey, err := protectorKey.CreateProtectedKey([]byte("primary-in"))
```

Comment:
maybe another variant where nil is passed, as this mimics what is done in `install.PrepareEncryptedSystemData`

Reply:
Good point. Done.